pidTuner.cpp

```diff
@@ -0,0 +1,109 @@
+#include "pidTuner.hpp"
+#include <iostream>
+#include <math.h>
+
+
+pidTuner::pidTuner(float ip, float ii, float id, float Sp, float Si, float Sd){
+    _initial_pid = pid_set(ip,ii,id);
+    _current_pid =  _initial_pid;
+    _hill_vector = pid_set(1,1,1);
+
+    _cycles = 0;
+    _test_num = 0;
+
+    _pScale = Sp;
+    _iScale = Si;
+    _dScale = Sd;
+
+    _overScale = 1;
+
+    _threshold = .5;
+}
+
+pidTuner::pid_set::pid_set(float ip, float ii, float id){
+    p=ip;
+    i=ii;
+    d=id;
+    score=0;
+}
+
+void pidTuner::start_cycle(){
+    //determine values to test
+    if(_test_num==_test_sets.size()){
```

Comment:
It checks if all of the values of the test_set have been tested. If they have it checks if it should keep testing and then generates new values to test. If not, it goes on to the next value to test.